galsim/interpolatedimage.py

```diff
@@ -966,8 +961,14 @@ def withGSParams(self, gsparams=None, **kwargs):
     def _sbp(self):
         stepk_image = self.stepk / self.kimage.scale  # usually 1, but could be larger
         with convert_cpp_errors():
+            # C++ layer needs Bounds that look like 0, N/2, -N/2, N/2-1
+            # So find the biggest N that works like that.
```

Comment:
Sounds like this leaves a possibility of truncation, is that right?  Should we warn in that case?

Reply:
This is what the old code did implicitly when copying to the KTable in C++.  But honestly, I don't think it's really ever an issue for the kinds of ways we use InterpolatedKImage.  The edges are usually designed to be negligible.  The only bit of potential information loss is if the input x dimension is even, in which case there is one more x value on the negative side than the positive side.  And this throws that column away.  We used to do that implicitly in the C++ code, since we only copied over the +x values to the KTable, so we missed that outermost -x column.